File: ScalaQuickStart/src/main/java/com/tirthal/learning/java2scala/javaway/controlstruct/ConditionalsSyntax.java
package com.tirthal.learning.java2scala.javaway.controlstruct;

// Syntax of conditional control structures in Java
public class ConditionalsSyntax {

	@SuppressWarnings("unused")
	public static void main(String[] args) {
		
		// (1) ---> if:then - it's a statement in Java, which means it carries out an action
		if (args.length==0)
			System.out.println("Arguments are null");
		else
			System.out.println("Arguments are not null");
		
		// (2) ---> Ternary (?:) - is an expressions in Java, which means it returns a value
		System.out.println(args.length==0 ? "Arguments are null" : "Arguments are not null");
		
		// (3) ---> Switch statement in Java - can match only primitives, enums, strings
		String day = "Monday";
		String dayType = "???"; 
		switch(day) {
			case "Monday":
			case "Tuesday":
			case "Wednesday":
			case "Thursday":
			case "Friday":
				dayType = "working day";
				break;
			case "Saturday":
			case "Sunday":
				dayType = "holiday";
				break;
			default:
				dayType = "unknown";
				break;
		}
		System.out.println(day + " is " + dayType);
		
	}
}

File: ScalaQuickStart/src/main/scala/com/tirthal/learning/java2scala/scalaway/controlstruct/ConditionalsSyntax.scala
package com.tirthal.learning.java2scala.scalaway.controlstruct

// Syntax of conditional control structures in Scala
object ConditionalsSyntax { 
  
  def main(args: Array[String]) {
    
    // (1) ---> if:then - it's an expression in Scala, which means it returns a value
    println(if(args.length==0) "Arguments are null" else "Arguments are not null")
    
    // (2) ---> Scala has no ternary (?:), because Scala ifs expression is equivalent to ternary
    // In face anything in Scala is an expression, even { ... } can return something
   
    // (3) ---> Scala has no Switch statement, instead it uses Match Expressions / Pattern Matching - It can match anything including even objects
    val day = "Monday"
    println(day + " is " + 
      (day match {
        case "Monday" | "Tuesday" | "Wednesday" | "Thurday" | "Friday" => "working day"
        case "Saturday" | "Sunday" => "holiday"
        case _ => "unknown"
      })
    )

  }
  
}